<ui version="4.0" >
 <class>MeshGui::DlgRegularSolid</class>
 <widget class="QDialog" name="MeshGui::DlgRegularSolid" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>265</width>
    <height>293</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>Primitives</string>
  </property>
  <property name="sizeGripEnabled" >
   <bool>true</bool>
  </property>
  <layout class="QGridLayout" >
   <property name="margin" >
    <number>9</number>
   </property>
   <property name="spacing" >
    <number>6</number>
   </property>
   <item row="0" column="0" >
    <widget class="QGroupBox" name="groupBox1" >
     <property name="title" >
      <string>Solid:</string>
     </property>
     <layout class="QGridLayout" >
      <property name="margin" >
       <number>9</number>
      </property>
      <property name="spacing" >
       <number>6</number>
      </property>
      <item row="1" column="0" >
       <widget class="QStackedWidget" name="widgetStack2" >
        <property name="currentIndex" >
         <number>0</number>
        </property>
        <widget class="QWidget" name="page" >
         <layout class="QGridLayout" >
          <property name="margin" >
           <number>9</number>
          </property>
          <property name="spacing" >
           <number>6</number>
          </property>
          <item row="0" column="0" >
           <layout class="QGridLayout" >
            <property name="margin" >
             <number>0</number>
            </property>
            <property name="spacing" >
             <number>6</number>
            </property>
            <item row="0" column="1" >
             <widget class="QDoubleSpinBox" name="boxLength_2" >
              <property name="maximum" >
               <double>1000.000000000000000</double>
              </property>
              <property name="minimum" >
               <double>0.000000000000000</double>
              </property>
              <property name="value" >
               <double>10.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="1" column="0" >
             <widget class="QLabel" name="textLabel3_2" >
              <property name="text" >
               <string>Width:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0" >
             <widget class="QLabel" name="textLabel2_2" >
              <property name="text" >
               <string>Length:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1" >
             <widget class="QDoubleSpinBox" name="boxWidth_2" >
              <property name="maximum" >
               <double>1000.000000000000000</double>
              </property>
              <property name="value" >
               <double>10.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="0" >
           <spacer>
            <property name="orientation" >
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" >
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="WStackPage1" >
         <layout class="QGridLayout" >
          <property name="margin" >
           <number>9</number>
          </property>
          <property name="spacing" >
           <number>6</number>
          </property>
          <item row="1" column="0" >
           <spacer>
            <property name="orientation" >
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType" >
             <enum>QSizePolicy::Expanding</enum>
            </property>
            <property name="sizeHint" >
             <size>
              <width>20</width>
              <height>51</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="0" >
           <layout class="QGridLayout" >
            <property name="margin" >
             <number>0</number>
            </property>
            <property name="spacing" >
             <number>6</number>
            </property>
            <item row="2" column="0" >
             <widget class="QLabel" name="textLabel4" >
              <property name="text" >
               <string>Height:</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1" >
             <widget class="QDoubleSpinBox" name="boxHeight" >
              <property name="maximum" >
               <double>1000.000000000000000</double>
              </property>
              <property name="value" >
               <double>10.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="0" column="1" >
             <widget class="QDoubleSpinBox" name="boxLength" >
              <property name="maximum" >
               <double>1000.000000000000000</double>
              </property>
              <property name="minimum" >
               <double>0.000000000000000</double>
              </property>
              <property name="value" >
               <double>10.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="0" column="0" >
             <widget class="QLabel" name="textLabel2" >
              <property name="text" >
               <string>Length:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1" >
             <widget class="QDoubleSpinBox" name="boxWidth" >
              <property name="maximum" >
               <double>1000.000000000000000</double>
              </property>
              <property name="value" >
               <double>10.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="1" column="0" >
             <widget class="QLabel" name="textLabel3" >
              <property name="text" >
               <string>Width:</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="WStackPage2" >
         <layout class="QGridLayout" >
          <property name="margin" >
           <number>9</number>
          </property>
          <property name="spacing" >
           <number>6</number>
          </property>
          <item row="1" column="0" >
           <widget class="QCheckBox" name="cylinderClosed" >
            <property name="text" >
             <string>Closed</string>
            </property>
            <property name="checked" >
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="0" >
           <spacer>
            <property name="orientation" >
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType" >
             <enum>QSizePolicy::Expanding</enum>
            </property>
            <property name="sizeHint" >
             <size>
              <width>31</width>
              <height>81</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="0" >
           <layout class="QGridLayout" >
            <property name="margin" >
             <number>0</number>
            </property>
            <property name="spacing" >
             <number>6</number>
            </property>
            <item row="0" column="0" >
             <widget class="QLabel" name="textLabel5" >
              <property name="text" >
               <string>Radius:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1" >
             <widget class="QDoubleSpinBox" name="cylinderRadius" >
              <property name="value" >
               <double>2.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="1" column="1" >
             <widget class="QDoubleSpinBox" name="cylinderLength" >
              <property name="value" >
               <double>10.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="1" column="0" >
             <widget class="QLabel" name="textLabel6" >
              <property name="text" >
               <string>Length:</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="WStackPage3" >
         <layout class="QGridLayout" >
          <property name="margin" >
           <number>9</number>
          </property>
          <property name="spacing" >
           <number>6</number>
          </property>
          <item row="1" column="0" >
           <widget class="QCheckBox" name="coneClosed" >
            <property name="text" >
             <string>Closed</string>
            </property>
            <property name="checked" >
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="0" >
           <layout class="QGridLayout" >
            <property name="margin" >
             <number>0</number>
            </property>
            <property name="spacing" >
             <number>6</number>
            </property>
            <item rowspan="2" row="2" column="0" >
             <widget class="QLabel" name="textLabel11" >
              <property name="text" >
               <string>Length:</string>
              </property>
             </widget>
            </item>
            <item rowspan="2" row="1" column="1" >
             <widget class="QDoubleSpinBox" name="coneRadius2" >
              <property name="value" >
               <double>4.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="3" column="1" >
             <widget class="QDoubleSpinBox" name="coneLength" >
              <property name="value" >
               <double>10.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="0" column="1" >
             <widget class="QDoubleSpinBox" name="coneRadius1" >
              <property name="value" >
               <double>2.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="0" column="0" >
             <widget class="QLabel" name="textLabel9" >
              <property name="text" >
               <string>Radius 1:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0" >
             <widget class="QLabel" name="textLabel10" >
              <property name="text" >
               <string>Radius 2:</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="2" column="0" >
           <spacer>
            <property name="orientation" >
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType" >
             <enum>QSizePolicy::Expanding</enum>
            </property>
            <property name="sizeHint" >
             <size>
              <width>31</width>
              <height>91</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="WStackPage4" >
         <layout class="QGridLayout" >
          <property name="margin" >
           <number>9</number>
          </property>
          <property name="spacing" >
           <number>6</number>
          </property>
          <item row="1" column="0" >
           <spacer>
            <property name="orientation" >
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType" >
             <enum>QSizePolicy::Expanding</enum>
            </property>
            <property name="sizeHint" >
             <size>
              <width>21</width>
              <height>151</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="0" >
           <layout class="QHBoxLayout" >
            <property name="margin" >
             <number>0</number>
            </property>
            <property name="spacing" >
             <number>6</number>
            </property>
            <item>
             <widget class="QLabel" name="textLabel14" >
              <property name="text" >
               <string>Radius:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="sphereRadius" >
              <property name="value" >
               <double>5.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="WStackPage5" >
         <layout class="QGridLayout" >
          <property name="margin" >
           <number>9</number>
          </property>
          <property name="spacing" >
           <number>6</number>
          </property>
          <item row="1" column="0" >
           <spacer>
            <property name="orientation" >
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType" >
             <enum>QSizePolicy::Expanding</enum>
            </property>
            <property name="sizeHint" >
             <size>
              <width>31</width>
              <height>81</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="0" >
           <layout class="QGridLayout" >
            <property name="margin" >
             <number>0</number>
            </property>
            <property name="spacing" >
             <number>6</number>
            </property>
            <item row="1" column="1" >
             <widget class="QDoubleSpinBox" name="ellipsoidRadius2" >
              <property name="value" >
               <double>4.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="1" column="0" >
             <widget class="QLabel" name="textLabel17" >
              <property name="text" >
               <string>Radius 2:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1" >
             <widget class="QDoubleSpinBox" name="ellipsoidRadius1" >
              <property name="value" >
               <double>2.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="0" column="0" >
             <widget class="QLabel" name="textLabel16" >
              <property name="text" >
               <string>Radius 1:</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="WStackPage6" >
         <layout class="QGridLayout" >
          <property name="margin" >
           <number>9</number>
          </property>
          <property name="spacing" >
           <number>6</number>
          </property>
          <item row="1" column="0" >
           <spacer>
            <property name="orientation" >
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType" >
             <enum>QSizePolicy::Expanding</enum>
            </property>
            <property name="sizeHint" >
             <size>
              <width>20</width>
              <height>91</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="0" >
           <layout class="QGridLayout" >
            <property name="margin" >
             <number>0</number>
            </property>
            <property name="spacing" >
             <number>6</number>
            </property>
            <item row="1" column="0" >
             <widget class="QLabel" name="textLabel20" >
              <property name="text" >
               <string>Radius 2:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0" >
             <widget class="QLabel" name="textLabel19" >
              <property name="text" >
               <string>Radius 1:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1" >
             <widget class="QDoubleSpinBox" name="toroidRadius2" >
              <property name="value" >
               <double>2.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="0" column="1" >
             <widget class="QDoubleSpinBox" name="toroidRadius1" >
              <property name="value" >
               <double>10.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item row="0" column="0" >
       <widget class="QComboBox" name="comboBox1" >
        <item>
         <property name="text" >
          <string>Plane</string>
         </property>
        </item>
        <item>
         <property name="text" >
          <string>Cube</string>
         </property>
        </item>
        <item>
         <property name="text" >
          <string>Cylinder</string>
         </property>
        </item>
        <item>
         <property name="text" >
          <string>Cone</string>
         </property>
        </item>
        <item>
         <property name="text" >
          <string>Sphere</string>
         </property>
        </item>
        <item>
         <property name="text" >
          <string>Ellipsoid</string>
         </property>
        </item>
        <item>
         <property name="text" >
          <string>Torus</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0" >
    <layout class="QHBoxLayout" >
     <property name="margin" >
      <number>0</number>
     </property>
     <property name="spacing" >
      <number>6</number>
     </property>
     <item>
      <spacer>
       <property name="orientation" >
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeType" >
        <enum>QSizePolicy::Expanding</enum>
       </property>
       <property name="sizeHint" >
        <size>
         <width>20</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="createSolidButton" >
       <property name="text" >
        <string>&amp;Create</string>
       </property>
       <property name="shortcut" >
        <string>Alt+C</string>
       </property>
       <property name="autoDefault" >
        <bool>true</bool>
       </property>
       <property name="default" >
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="buttonClose" >
       <property name="text" >
        <string>Cl&amp;ose</string>
       </property>
       <property name="shortcut" >
        <string>Alt+O</string>
       </property>
       <property name="autoDefault" >
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11" />
 <pixmapfunction>qPixmapFromMimeSource</pixmapfunction>
 <tabstops>
  <tabstop>comboBox1</tabstop>
  <tabstop>boxLength</tabstop>
  <tabstop>boxWidth</tabstop>
  <tabstop>boxHeight</tabstop>
  <tabstop>cylinderRadius</tabstop>
  <tabstop>cylinderLength</tabstop>
  <tabstop>cylinderClosed</tabstop>
  <tabstop>coneRadius1</tabstop>
  <tabstop>coneRadius2</tabstop>
  <tabstop>coneLength</tabstop>
  <tabstop>coneClosed</tabstop>
  <tabstop>sphereRadius</tabstop>
  <tabstop>ellipsoidRadius1</tabstop>
  <tabstop>ellipsoidRadius2</tabstop>
  <tabstop>toroidRadius1</tabstop>
  <tabstop>toroidRadius2</tabstop>
  <tabstop>createSolidButton</tabstop>
  <tabstop>buttonClose</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>comboBox1</sender>
   <signal>activated(int)</signal>
   <receiver>widgetStack2</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>74</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel" >
     <x>230</x>
     <y>116</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonClose</sender>
   <signal>clicked()</signal>
   <receiver>MeshGui::DlgRegularSolid</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>222</x>
     <y>284</y>
    </hint>
    <hint type="destinationlabel" >
     <x>52</x>
     <y>296</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
